src/box/lua/load_cfg.lua

```diff
@@ -259,6 +250,36 @@ local template_cfg = {
     txn_timeout           = 'number',
 }
 
+local box2log_map = {
+    log = 'log',
+    log_nonblock = 'nonblock',
+    log_level = 'level',
+    log_format = 'format',
+}
```

Comment:
It came from the request where `log` module need to be used _before_ `box.cfg` call. IOW customers wanted to setup log module, use it for some time and only later call `box.cfg`. Due to this I've had to implement a way more complex mappings as it should be (simply because logging never had a proper architecture but been bound to box.cfg).

Reply:
@locker Well it makes sense to me. This works while log module is configured only as part of the box configuration. Otherwise log (and other modules) have to keep adapters for every user. So typically approach when user cares of adapters looks more managable.

Anyway I can keep the original approach and still substantially reduce complexity.

Reply:
Overall, I'm not sure which approach is better. The one thing that I liked about the old approach is that it kept `load_cfg` infrastructure simple by offloading all the complexity to modules. There were more code, but interaction between different subsystems were clearer IMO. I wouldn't bother reworking it unless it's necessary to fix a bug or implement a new feature. I won't block it, but I won't approve it, either, unless you get an approval from a second reviewer from our team, preferably, from @alyapunov.

Reply:
Ok. I'm making some changes after the review now. So let's check second version.